### PR TITLE
Add pdf reports, markdown overview and analyses for the first basic benchmark done on my pc

### LINKED ISSUE #34: Add initial reports and analyses

### PR DESCRIPTION
**Pull request by GitGab19**
Closes #34

### CODE REVIEW

**Review by Shourya742 — CHANGES_REQUESTED** [review]
Completed the initial reading. I will proofread soon. This review has clarified the benchmarking tool's overview and experimental metrics.

File: docs/benchmarking-tool-overview.md [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+    * <span style="text-decoration:underline;">SV1 shares</span>
+        * _Description_: it tracks metrics about shares submitted by the miner(s) used to benchmark SV1
+        * _Data Collection Method_: a SV1 custom proxy located between the miner and the SV1 Pool forwards all the traffic and analyze it, extracting the shares metrics
+        * _Data Analysis and usage_: shares are analyzed and tracked as
+            * number of submitted shares
+            * number of valid shares
+            * number of stale shares
+            * acceptance rate percentage
+    * <span style="text-decoration:underline;">SV1 mined blocks</span>
+        * _Description_: it tracks metrics about blocks mined by the miner(s) used to benchmark SV1
```
Comment (marathon-gary) [inline docs/benchmarking-tool-overview.md:62]:
```suggestion
        * _Description_: Tracks metrics about blocks mined by the miner(s) used to benchmark SV1
```

File: docs/benchmarking-tool-overview.md [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+## Metrics analyzed 📐
+
+Following the grafana dashboard structure, which wraps all the metrics currently captured by the benchmarking tool, here’s the list of measurements, with related explanation.
+
+
+
+* **Shares, blocks and templates stats**
+    * <span style="text-decoration:underline;">SV1 shares</span>
+        * _Description_: it tracks metrics about shares submitted by the miner(s) used to benchmark SV1
+        * _Data Collection Method_: a SV1 custom proxy located between the miner and the SV1 Pool forwards all the traffic and analyze it, extracting the shares metrics
```
[the remark this answers is not in the document]
Comment (marathon-gary) [reply docs/benchmarking-tool-overview.md:55]: `A SV1...` is grammatically correct.

File: docs/benchmarking-tool-overview.md [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+            * number of submitted shares
+            * number of valid shares
+            * number of stale shares
+            * acceptance rate percentage
+    * <span style="text-decoration:underline;">SV1 mined blocks</span>
+        * _Description_: it tracks metrics about blocks mined by the miner(s) used to benchmark SV1
+        * _Data Collection Method_: a SV1 custom proxy located between the Bitcoin node and the SV1 Pool forwards all the traffic and analyzes it, extracting the blocks submissions to the network
+        * _Data Analysis and usage_: blocks mined are added to a counter
+    * <span style="text-decoration:underline;">SV2 shares</span>
+        * _Description_: it tracks metrics about shares submitted by the miner(s) used to benchmark SV2
```
Comment (marathon-gary) [inline docs/benchmarking-tool-overview.md:66]:
```suggestion
        * _Description_: Tracks shares submitted by the miner(s) used to benchmark SV2
```

File: docs/benchmarking-tool-overview.md [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+        * _Data Collection Method_: a SV2 custom proxy located between the miner and the SV2 Pool or SV2 Job Declarator Client (JDC) forwards all the traffic and analyze it, extracting the shares metrics
+        * _Data Analysis and usage_: shares are analyzed and tracked as
+            * number of submitted shares
+            * number of valid shares
+            * number of stale shares
+            * acceptance rate percentage  
+    * <span style="text-decoration:underline;">SV2 mined blocks</span>
+        * _Description_: it tracks metrics about blocks mined by the miner(s) used to benchmark SV2
```
Comment (marathon-gary) [inline docs/benchmarking-tool-overview.md:74]:
```suggestion
        * _Description_: Tracks metrics about blocks mined by the miner(s) used to benchmark SV2
```

**Comment by marathon-gary** [comment]
PDF generation of the overview should be in a README, imho. PDF won't age well unless we have some CI/CD that generates when commits are added. just my two sats

**Comment by GitGab19** [comment]
> PDF generation of the overview should be in a README, imho. PDF won't age well unless we have some CI/CD that generates when commits are added. just my two sats

Could you elaborate more?

**Comment by marathon-gary** [comment]
> > PDF generation of the overview should be in a README, imho. PDF won't age well unless we have some CI/CD that generates when commits are added. just my two sats
> 
> Could you elaborate more? Should we label them with tool version? So that it will be clear in the future for everyone (?)

Originally my comment was intended to deter having a PDF in the repo and instead to just provide instructions for generating a PDF, however, after reading it through more comprehensively, I think the PDF is fine to stay as an example and initial documentation of findings.

**Review by marathon-gary — APPROVED** [review]

**Review by Shourya742 — APPROVED** [review]
ACK 4bbd76664c71d9a10a426fa8cfcd10aff632c737

**Pull request merged**